app/src/main/java/com/flowfoundation/wallet/page/restore/keystore/viewmodel/KeyStoreRestoreViewModel.kt

```diff
@@ -1124,18 +1125,34 @@ class KeyStoreRestoreViewModel : ViewModel() {
         ioScope {
             val deviceInfoRequest = DeviceInfoManager.getDeviceInfoRequest()
             val service = retrofit().create(ApiService::class.java)
-            val request = RegisterRequest(
-                username = username,
+
+            // Get Firebase JWT for signing (v4 requires signature verification)
+            val firebaseJwt = getFirebaseJwt()
+            logd("KeyStoreRestoreViewModel", "Got Firebase JWT for v4 registration")
+
+            // Sign the Firebase JWT with the crypto provider
+            val signature = cryptoProvider.getUserSignature(firebaseJwt)
+            logd("KeyStoreRestoreViewModel", "Signed Firebase JWT, signature length: ${signature.length}")
+
+            // Create v4 registration request with FlowAccountInfo containing signature
+            val flowAccountInfo = FlowAccountInfo(
                 accountKey = AccountKey(
                     publicKey = cryptoProvider.getPublicKey(),
                     signAlgo = cryptoProvider.getSignatureAlgorithm().cadenceIndex,
                     hashAlgo = cryptoProvider.getHashAlgorithm().cadenceIndex,
                 ),
+                signature = signature
+            )
+
+            val request = RegisterRequest(
+                flowAccountInfo = flowAccountInfo,
+                evmAccountInfo = null, // No EVM account for now
```

Comment:
There is EVMAccountInfo for keystore type

Reply:
And what about the private key import ? I believe it might also need the same handling as well